### PR TITLE
Updated HelpPageAreaRegistration.cs

### PR DESCRIPTION
**Pull request by adamcarr1976**
- Altered HelpPage Default route to use a generic id parameter, this renders the hyperlinks with query string parameters.

Addresses #131

### CODE REVIEW

**Comment by adamcarr1976** [comment]
I think this is a routing issue because the last segment of the URI contains a period.  This is being handled by the web server and it interprets the period as a file type.  It passes off the request to be resolved to the relative handler; which in this case does not exist, hence the 404.  

The following test works fine, which should hopefully prove this is the case.
http://localhost:nnnn/Help/Api?apiId=POST-Start-A.B-End-Do

The safest change would be to alter the default routing configuration.  So the apiId is passed as an actual query string parameter.

**Comment by dougbu** [comment]
This PR does not work end-to-end. Please have a look at the following unit test failure. (Running all tests in VS should just work.)
> `WebApiHelpPageWebHost.UnitTest.HelpPageAreaRegistrationTest.RegisterArea [FAIL]`

File: src/WebApiHelpPage/Areas/HelpPage/HelpPageAreaRegistration.cs [hunk 1]
```diff
@@ -17,10 +17,10 @@ public override void RegisterArea(AreaRegistrationContext context)
         {
             context.MapRoute(
                 "HelpPage_Default",
-                "Help/{action}/{apiId}",
-                new { controller = "Help", action = "Index", apiId = UrlParameter.Optional });
+                "Help/{action}/{id}",
+                new { controller = "Help", action = "Index", id = UrlParameter.Optional });
```
[the remark this answers is not in the document]
Comment (dougbu) [reply src/WebApiHelpPage/Areas/HelpPage/HelpPageAreaRegistration.cs]: I suspect the root cause is in how the `apiId`s are generated: https://github.com/aspnet/AspNetWebStack/blob/62d0b2df0bcca848a8e0848fd1866928df15528e/src/WebApiHelpPage/Areas/HelpPage/ApiDescriptionExtensions.cs#L31
and
https://github.com/aspnet/AspNetWebStack/blob/62d0b2df0bcca848a8e0848fd1866928df15528e/src/WebApiHelpPage/VB/Areas/HelpPage/ApiDescriptionExtensions.vb#L32

Put another way, the issue isn't in how `apiId`s are used but in how they are generated by the `ApiDescriptionExtensions.GetFriendlyId(...)` extension methods. A number of characters meaningful to routing aren't handled. This includes at least `.`, `=`, `*`, and  `:`. 

To be completely correct, should probably remove every character from a `:` to the end of the constraint e.g. `/{id:int}` -> `-id`.

[the remark this answers is not in the document]
Comment (dougbu) [reply src/WebApiHelpPage/Areas/HelpPage/HelpPageAreaRegistration.cs]: @rynowak realizing almost any literal character can separate template segments, what say you about flipping `GetFriendlyId(...)` around? That is, extract alpha-numeric characters versus remove a few known-problematic characters.

Could instead stick with the current approach but handle the remaining characters special-cased in template. Did I miss any characters? And, given these helpers are best-effort, worth removing the constraints?

**Comment by dougbu** [comment]
Thank you for your proposal. We're closing this issue as no updates have been provided in a timely manner.

If you wish to fix #131 as I suggested, please open a new PR that addresses my comments above.

**Pull request closed without merging**